nccl/common/prep_ami.sh

```diff
@@ -0,0 +1,225 @@
+    echo "export LD_LIBRARY_PATH=${HOME}/rdma-core/build/lib/:\$LD_LIBRARY_PATH" >> ~/.bashrc
+    source ~/.bash_profile
+}
+
+install_libfabric() {
+
+    cd ${HOME}/libfabric
+    ./autogen.sh
+    ./configure --prefix=${HOME}/libfabric/install/ \
+         --enable-debug  \
+         --enable-mrail  \
+         --enable-tcp    \
+         --enable-rxm    \
+         --disable-rxd   \
+         --disable-verbs \
+         --enable-efa=${HOME}/rdma-core/build
+    make -j 4
+    make install
+    echo "export LD_LIBRARY_PATH=${HOME}/libfabric/install/lib/:\$LD_LIBRARY_PATH" >> ~/.bash_profile
+    echo "export LD_LIBRARY_PATH=${HOME}/libfabric/install/lib/:\$LD_LIBRARY_PATH" >> ~/.bashrc
+    source ~/.bash_profile
+}
+
+prepare_libfabric_without_pr() {
+
+    echo "==> Building libfabric"
+    cd ${HOME}
+    sudo rm -rf libfabric
+    git clone https://github.com/ofiwg/libfabric -b 'master'
+}
+
+prepare_libfabric_with_pr() {
+
+    echo "==> This PR belongs to lifabric repo: ofiwg/libfabric"
+    echo "==> Starting custom libfabric installation"
+    # Pulls the libfabric repository and checks out the pull request commit
+    cd ${HOME}
+    sudo rm -rf libfabric
+    git clone https://github.com/ofiwg/libfabric
+    cd ${HOME}/libfabric
+    git fetch origin +refs/pull/$PULL_REQUEST_ID/*:refs/remotes/origin/pr/$PULL_REQUEST_ID/*
+    git checkout $PULL_REQUEST_REF -b PRBranch
+}
+
+check_efa_installation_libfabric(){
+
+    echo "==> Check if the EFA installed correctly"
+    ${HOME}/libfabric/install/bin/fi_info -p efa
+}
+
+install_nccl() {
+
+    echo "==> Install NCCL"
+    cd $HOME
+    sudo rm -rf nccl
+    sudo git clone https://github.com/NVIDIA/nccl.git && cd nccl
+    sudo make -j src.build CUDA_HOME=${latest_cuda}
+}
+
+install_nccl_tests() {
+
+    echo "==> Install NCCL Tests"
+    cd $HOME
+    sudo rm -rf nccl-tests
+    git clone https://github.com/NVIDIA/nccl-tests.git && cd nccl-tests
+    make MPI=1 MPI_HOME=$HOME/anaconda3 NCCL_HOME=$HOME/nccl/build CUDA_HOME=${latest_cuda}
+}
+
+install_aws_ofi_nccl_plugin() {
+
+    cd $HOME/aws-ofi-nccl
+    ./autogen.sh
+    ./configure --prefix=$HOME/aws-ofi-nccl/install \
+                --with-mpi=$HOME/anaconda3 \
+                --with-libfabric=$HOME/libfabric/install \
+                --with-nccl=$HOME/nccl/build \
+                --with-cuda=${latest_cuda}
+    make && make install
+}
+
+prepare_aws_ofi_nccl_plugin_without_pr() {
+
+    echo "==> Install aws-ofi-nccl plugin"
+    echo "==> Configure from branch: ${plugin_branch} provider: ${PROVIDER}"
+    cd $HOME
+    sudo rm -rf aws-ofi-nccl
+    git clone https://github.com/aws/aws-ofi-nccl.git -b ${plugin_branch}
+}
+
+prepare_aws_ofi_nccl_plugin_with_pr() {
+
+    echo "==> This PR belongs to nccl repo: aws/aws-ofi-nccl"
+    echo "==> Install aws-ofi-nccl plugin"
+    cd $HOME
+    sudo rm -rf aws-ofi-nccl
+    if [[ ${TARGET_BRANCH} == 'master' && ${PROVIDER} == 'tcp;ofi_rxm' ]]; then
```

Comment:
We can simplify these branches with a bash map for target branch and provider instead and one set of git clone/fetch/checkout commands.

Reply:
ack.